design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,349 @@
+
+Following is an example CR for the cluster wide CRD `NodeNetworkIPPool`:
+
+```yaml
+apiVersion: nmstate.io/v1
+kind: NodeNetworkIPPool
+metadata:
+  name: traffic-1
+spec:
+  range:
+    cidr: 10.10.10.0/24
+    start: 1
+    end: 10
+    exclude: [4, 7]
+status:
+  allocated-offsets: [1, 2, 5, 6, 8]
+  conditions:
+  - type: Available
+    status: True
+    reason: ReconciledPool
+    message: The pool is up to date
+  - type: Failure
+    status: False
+    reason: ReconciledPool
+apiVersion: nmstate.io/v1
+kind: NodeNetworkIPPool
+metadata:
+  name: traffic-2
+spec:
+  range:
+    cidr: 10.10.11.0/24
+    start: 1
+    end: 10
+    exclude: [4, 7]
+status:
+  allocated-offsets: [1, 2, 5, 6, 8]
+  conditions:
+  - type: Available
+    status: True
+    reason: ReconciledPool
+    message: The pool is up to date
+  - type: Failure
+    status: False
+    reason: ReconciledPool
+```
+
+The resource does not have a namespace since it's cluster wide information
+similar to NNCP. The `spec.range.cidr` is an IP range on CIDR format to
+configure IP pool, `spec.range.start` is the begining of the range and
+`spec.range.end` is the end, there is also an `spec.range.exclude` list of
+offset that will not be allocated.
+
+The `status.allocated-offsets` store the allocated IPs represented as a
+list of offsets from the range.
+
+Reseting `status.allocated-offset` field will trigger a Reconcile on the
+`NodeNetworkIPPool` and `allocated-offsets` will be re-calculated.
+In this process the reason will be "ReconcilingPool" and `Available`
+marked as false, so users can wait on it.
+
+```yaml
+  conditions:
+  - type: Available
+    status: False
+    reason: ReconcilingPool
+    message: The pool is being updated
+  - type: Failure
+    status: False
+    reason: ReconcilingPool
+```
+
+To assign an IP address from a pool to an interface using NNCP, a special
+capture syntax is introduced understood by kubernetes-nmstate
+`ippool.[name].allocate` that will trigger the ip allocation mechanism,
+it will inject the result on the captured state passed to nmpolicy so it can be
+consumed by it, the `ippool.[name].allocate` tag will return a struct with the
+foollowing fields `ip` and `prefix-length`.
+
+```yaml
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: static-ip
+capture:
+    address-from-traffic-1: ippool.traffic-1.allocate
+    address-from-traffic-2: ippool.traffic-2.allocate
+desiredState:
+    interfaces:
+    - name: eth1
+      type: ethernet
+      state: up
+      ipv4:
+        address:
+        - ip: "{{ capture.address-from-traffic-1.ip }}"
+          prefix-length: "{{ capture.address-from-traffic-2.prefix-length }}"
+        dhcp: false
+        enabled: true
+    - name: eth2
+      type: ethernet
+      state: up
+      ipv4:
+        address:
+        - ip: "{{ capture.address-from-traffic-1.ip }}"
+          prefix-length: "{{ capture.address-from-traffic-2.prefix-length }}"
```

Comment:
I think your `eth1` should get address / prefix length from pool 1, and `eth2` get address / prefix length from pool 2 .